autosklearn/automl.py

```diff
@@ -931,23 +931,49 @@ def subsample_if_too_large(
                     new_num_samples,
                 )
                 if task in CLASSIFICATION_TASKS:
-                    try:
+                    # Identify if it has unique labels and allow for
+                    # stratification, with unique labels in training set
+                    values, idxs, counts = np.unique(y, axis=0,
+                                                     return_index=True,
+                                                     return_counts=True)
+                    unique_labels = {
+                        idx: value
+                        for value, idx, count in zip(values, idxs, counts)
+                        if count == 1
+                    }
+
+                    # If there are unique labeled elements, remove them and
+                    # place them back in later
+                    if len(unique_labels) > 0:
+                        idxs_of_unique = np.asarray(list(unique_labels.keys()))
+                        unique_X = X[idxs_of_unique]
+                        unique_y = y[idxs_of_unique]
+
+                        # TODO optimization
```

Comment:
I like the comment, but I'm not sure if the TODO statement should stay here?

Reply:
Sorry, it's a habit to scatter `# TODO context` whereever there's some internal code point to make as I have some bindings to find them quickly. Would you be okay with NOTE instead?
```
# NOTE optimization
#    ....
```

Reply:
Yes, that'd be better. TODOs are displayed prominently in pycharm, and this isn't really one...